lib/optimizely/event/batch_event_processor.rb

```diff
@@ -36,10 +36,11 @@ class BatchEventProcessor < EventProcessor
 
     def initialize(
       event_queue: SizedQueue.new(DEFAULT_QUEUE_CAPACITY),
-      event_dispatcher:,
+      event_dispatcher: Optimizely::EventDispatcher.new,
```

Comment:
Why are we adding a default value here?

Reply:
This is not related to LogEvent, therefore moved to PR: https://github.com/optimizely/ruby-sdk/pull/200